adxl345/adxl345.go

```diff
@@ -92,7 +92,7 @@ func (d *Device) Restart() {
 // it in µg (micro-gravity). When one of the axes is pointing straight to Earth
 // and the sensor is not moving the returned value will be around 1000000 or
 // -1000000.
-func (d *Device) ReadAcceleration() (x int32, y int32, z int32, err error) {
+func (d *Device) ReadAcceleration() (x int16, y int16, z int16, err error) {
```

Comment:
It would be better to not change the signature of the exported functions, so they are consistent with others devices in the drivers package.

Instead perhaps just use casting from the internal functions?

Reply:
Roger, done.

Reply:
Coming from a C world where an int was supposed to be  the best "native" length of a CPU. In go this seems to apply as-well [see](https://go.dev/tour/basics/11)

> The int, uint, and uintptr types are usually 32 bits wide on 32-bit systems and 64 bits wide on 64-bit systems. When you need an integer value you should use int unless you have a specific reason to use a sized or unsigned integer type. 

So I would assume that the signature is `int` but maybe this is due to the fact that there is support for the old 8 bit CPUS....

Back in the day when I did a bunch of DSP work, there was overhead when not using the native data type caused by shifting variables in registers. This may all fells apart when there are 8bit integers.